src/components/implementation/torrent/ramfs/ramfs.c

```diff
@@ -23,6 +24,43 @@ struct fsobj root;
 #define UNLOCK() if (lock_release(&fs_lock)) BUG();
 
 #define MIN_DATA_SZ 256
+#define DEBUG 0
+
+struct file_data {
+	cbuf_t cbuf_id;
+	char *chunk_start;
+	int cbuf_len;
+	int start;
+	int len;
+	void* next;
+};
+
+int file_data_init(struct file_data **fd, 
+		cbuf_t cbuf_id, 
+		char* chunk_start, 
+		int cbuf_length, 
+		int start, 
+		int len, 
+		void* next)
+{
+	assert(*fd);
+	(*fd)->cbuf_id = cbuf_id;
```

Comment:
That said,  I don't see why a `**fd` pointer is passed in rather than a `*fd`.  So I don't think that the argument is correct.